tests/test.js

```diff
@@ -0,0 +1,146 @@
+const Caballo = require("../src/caballo.js");
+const Jinete = require("../src/jinete.js");
+const emparejamiento = require("../src/emparejado.js");
+const crear_matriz_costo = require("../src/matriz_costos.js");
+
+function costo_emparejamiento(matriz, binomio) {
+    let costo = 0
+
+    for (let par of binomio) {
+        costo += matriz[par[1]][par[0]]
+    }
+    return costo
+}
+
+function comprobar_matriz(matriz, nfil, ncol) {
+    let menor_que_uno = false
+
+    for (let i = 0; i < nfil; i++) {
+        for (let j = 0; j < ncol; j++) {
+            if (matriz[i][j] < 1) {
+                menor_que_uno = true
+            }
+        }
+    }
+
+    return menor_que_uno
+}
+
+
+
+ //Creo unos caballos ejemplo
+ let caballo1 = new Caballo("A", 3);
+ let caballo2 = new Caballo("B", 2);
+ let caballo3 = new Caballo("C", 1);
+ let caballo4 = new Caballo("D", 0);
+
+ const horses = [caballo1, caballo2, caballo3, caballo4];
+
+ //Creo las preferencias de los alumnos a los caballos
+ let pref_marta = [
+     new Map([[caballo1, 3]]),
+     new Map([[caballo2, 3]]),
+     new Map([[caballo3, 0]]),
+     new Map([[caballo4, 0]]),
+ ];
+
+ let pref_carlos = [
+     new Map([[caballo1, 0]]),
+     new Map([[caballo2, 3]]),
+     new Map([[caballo3, 3]]),
+     new Map([[caballo4, 0]]),
+ ];
+
+ let pref_paco = [
+     new Map([[caballo1, 0]]),
+     new Map([[caballo2, 0]]),
+     new Map([[caballo3, 3]]),
+     new Map([[caballo4, 0]]),
+ ];
+
+ let pref_alberto = [
+     new Map([[caballo1, 0]]),
+     new Map([[caballo2, 0]]),
+     new Map([[caballo3, 0]]),
+     new Map([[caballo4, 3]]),
+ ];
+
+ //Creo uno jinetes ejemplo
+ let marta = new Jinete("Marta", 3, pref_marta);
+ let carlos = new Jinete("Carlos", 2, pref_carlos);
+ let paco = new Jinete("Paco", 1, pref_paco);
+ let alberto = new Jinete("Alberto", 0, pref_alberto);
+
+
+ const jinetes = [marta, carlos, alberto, paco];
+
+ let njinetes = jinetes.length
+ let ncaballos = horses.length
+ let matriz = crear_matriz_costo(jinetes, horses)
+
+ 
+// TESTS //
+
+describe("Pruebas de corrección", () => {
+    
+    test('El número de jinetes debe de ser menor o igual al de caballos', () => {
```

Comment:
Esta es una condición que tú estableces. ¿Por qué lo testeas?

Reply:
Es una restricción del propio problema. Viene detallada en el [README](README.md).

Reply:
Marta, que el valor de las dos variables *lo has puesto tú*. Si lo que quieres es testear la condición, tendrás que poner código que tire una excepción y comprobar que efectivamente se lanza, como, por cierto, has marcado aquí;:

>  ¿Se han testeado también las excepciones?

Y *no* has hecho. Os he repetido muchas veces que la lista de comprobación está ahí para que la sigáis y os ayude a superar el objetivo. Pero si no solo la marcas a pesar de no haberlo hecho, sino que además ni siquiera entiendes que tenías que hacerlo, es doble trabajo para mi y también para ti, que tendrás que hacerlo.